internal/services/network/virtual_network_resource.go

```diff
@@ -406,6 +429,16 @@ func expandVirtualNetworkProperties(ctx context.Context, d *pluginsdk.ResourceDa
 		}
 	}
 
+	if v, ok := d.GetOk("encryption"); ok {
+		if vList := v.([]interface{}); len(vList) > 0 && vList[0] != nil {
+			encryptionConf := vList[0].(map[string]interface{})
+			properties.Encryption = &network.VirtualNetworkEncryption{
+				Enabled:     pointer.To(true),
+				Enforcement: network.VirtualNetworkEncryptionEnforcement(encryptionConf["enforcement"].(string)),
+			}
+		}
```

Comment:
@wuxu92 how would users remove this value? We'd need to send `Encryption.enabled = false` when `len(vList) == 0`?

Reply:
the update method will set the `Encryption.Enabled` to false when leave the `Encryption` block as null(not set).
and I submitted a PR to update the acctest for this case: https://github.com/hashicorp/terraform-provider-azurerm/pull/22807.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/2633022/1a03b122-de35-485a-9df6-10740650258b)